detox/src/android/espressoapi/DetoxMatcher.js

```diff
@@ -14,39 +14,51 @@ function sanitize_matcher(matcher) {
   return originalMatcher.type ? originalMatcher.value : originalMatcher;
 } 
 class DetoxMatcher {
-  static matcherForText(text) {
+  static matcherForText(text, isRegex) {
     if (typeof text !== "string") throw new Error("text should be a string, but got " + (text + (" (" + (typeof text + ")"))));
+    if (typeof isRegex !== "boolean") throw new Error("isRegex should be a boolean, but got " + (isRegex + (" (" + (typeof isRegex + ")"))));
```

Comment:
Consider extracting these assertions into a separate method. Doing so would:
- Minimize code duplication
- Adhere to the Single Responsibility Principle (SRP)

Reply:
this file is autogenerated based on params on `DetoxMatcher.java`. Did you mean I should edit the generation code?

Reply:
ahh, right. never mind then 😆